src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c

```diff
@@ -31,49 +39,404 @@ const ActorInit Obj_Chan_InitVars = {
+    ObjChan* this = thisx;
+    ObjChan* temp;
+    s32 i;
+    Vec3f sp60;
+    Vec3s sp58;
+
+    if (this->unk1D0 > 0.0f) {
+        this->unk1D4 += 0x190;
+        if (this->unk1D0 <= 400.0f) {
+            this->unk1D0 = this->unk1D0 * 0.99f;
+        } else {
+            this->unk1D0 = this->unk1D0 - 1.0f;
+        }
+        if (this->unk1D0 < 0.0f) {
+            this->unk1D4 = 0;
+            this->unk1D0 = 0.0f;
+        }
+    }
+    this->actor.shape.rot.z = (Math_SinS(this->unk1D4) * this->unk1D0);
+    if ((this->stateFlags & OBJCHAN_STATE_START_CUTSCENE) && func_8013E2D4(&this->actor, this->cutscenes[0], -1, 0)) {
+        this->stateFlags |= OBJCHAN_STATE_CUTSCENE;
+        this->stateFlags &= ~OBJCHAN_STATE_START_CUTSCENE;
+    }
+    if ((this->stateFlags & OBJCHAN_STATE_CUTSCENE) && this->rotationSpeed == OBJCHAN_ROTATION_SPEED) {
+        this->stateFlags &= ~OBJCHAN_STATE_CUTSCENE;
+        ActorCutscene_Stop(this->cutscenes[0]);
+    }
+    Matrix_RotateY(this->actor.shape.rot.y, 0);
+    Matrix_InsertXRotation_s(this->actor.shape.rot.x, 1);
+    Matrix_InsertZRotation_s(this->actor.shape.rot.z, 1);
+    Matrix_RotateY(this->rotation, 1);
+    Matrix_GetStateTranslationAndScaledY(this->unk1CC, &this->actor.world.pos);
+    Math_Vec3f_Sum(&this->actor.world.pos, &this->unk1C0, &this->actor.world.pos);
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    for (i = 0; i < 5; i++) {
+        temp = this->pots[i];
+        if (temp != NULL) {
+            ObjChan_CalculatePotPosition(
+                &sp60, &sp58, &this->actor.world.pos, &this->actor.shape.rot,
+                (s16)((s32)((((f32)i * 360.0f) / 5.0f) * (65536.0f / 360.0f)) + this->rotation));
+            Math_Vec3f_Copy(&temp->actor.world.pos, (Vec3f*)&sp60);
+            Math_Vec3s_Copy(&temp->actor.shape.rot, &this->actor.shape.rot);
+            temp->actor.shape.rot.y = this->rotation;
+            Math_Vec3f_ToVec3s(&temp->collider.dim.pos, &temp->actor.world.pos);
+        }
+    }
+    if ((this->collider.base.acFlags & AC_HIT) && (this->collider.info.acHitInfo->toucher.dmgFlags & 0x800)) {
+        Actor_SetSwitchFlag(globalCtx, this->actor.params & 0x7F);
+    }
+    if (Flags_GetSwitch(globalCtx, this->actor.params & 0x7F)) {
+        if (!(this->stateFlags & OBJCHAN_STATE_FIRE_DELAY)) {
+            this->rotationSpeed = 0;
+            this->flameSize = 0.0f;
+            for (i = 0; i < 5; i++) {
+                temp = this->pots[i];
+                if (temp != NULL) {
+                    this->pots[i]->stateFlags |= OBJCHAN_STATE_FIRE_DELAY;
+                    this->pots[i]->fireDelayFrames = 20 + i * 5;
+                    this->pots[i]->flameSize = 0.0f;
+                }
+            }
+            this->stateFlags |= OBJCHAN_STATE_FIRE_DELAY;
+            this->stateFlags |= OBJCHAN_STATE_ON_FIRE;
+            this->stateFlags |= OBJCHAN_STATE_START_CUTSCENE;
+            return;
```

Comment:
Would adding an else after the outer most if statement take care of this return?